sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.childrenResolved) {
```

Comment:
In the above context, I confirmed that `encodeFor[T]` is not resolved and `encodeFor[T].deserializer` is resolved.
Do you want to see my log output?